js/main.js

```diff
@@ -1,36 +1,63 @@
+// global dom variables
 const $image = document.getElementById('img');
 const $photo = document.getElementById('img-placeholder');
+const $form = document.querySelector('#form');
+const $list = document.querySelector('#list');
+const $entryForm = document.getElementById('entry-form');
+const $entries = document.getElementById('entries');
+const $noEntries = document.getElementById('no-entries');
+const $anchorTag = document.querySelector('.anchor-tag');
+const $anchorTagTwo = document.querySelector('.new-button');
+const $formHeader = document.querySelector('.entry-header');
+
 $image.addEventListener('input', function (event) {
   $photo.setAttribute('src', event.target.value);
 }
 );
 
-const $form = document.querySelector('#form');
-
 $form.addEventListener('submit', function (event) {
   event.preventDefault();
   const messageData = {};
-  messageData.entryId = data.nextEntryId;
-  data.nextEntryId++;
-  messageData.title = event.target.elements.title.value;
-  messageData.img = event.target.elements.img.value;
-  messageData.notes = event.target.elements.notes.value;
-  data.entries.unshift(messageData);
-  $photo.setAttribute('src', 'images/placeholder-image-square.jpg');
-  $form.reset();
-  // updated code below
-  $list.prepend(renderEntry(messageData));
-  viewSwap('entries');
-  toggleNoEntries();
-
+  if (data.editing === null) {
+    messageData.entryId = data.nextEntryId;
+    data.nextEntryId++;
+    messageData.title = event.target.elements.title.value;
+    messageData.img = event.target.elements.img.value;
+    messageData.notes = event.target.elements.notes.value;
+    data.entries.unshift(messageData);
+    $photo.setAttribute('src', 'images/placeholder-image-square.jpg');
+    $form.reset();
+    // updated code below
+    $list.prepend(renderEntry(messageData));
+    $form.reset();
+    viewSwap('entries');
+    toggleNoEntries();
+
+    // 3rd edit below, this is to fill out form template save edit information on submit
+  } else {
+    data.editing.title = event.target.elements.title.value;
+    data.editing.img = event.target.elements.img.value;
+    data.editing.notes = event.target.elements.notes.value;
```

Comment:
You shouldn't be updating `data.editing` here, this should be happening somewhere else. You should be assigning these values to the `messageData` object.